smartcontractkit/external-adapters-js
# Fix/NCFX optional forex endpoint

## Closes [PDI-181](https://smartcontract-it.atlassian.net/browse/PDI-181)

## Description

Made the env vars required for the forex endpoint optional since not all NOPs have credentials to use it.

## Changes

- Made `FOREX_WS_USERNAME` and `FOREX_WS_PASSWORD` optional
- Bumped framework version to use the custom input validation feature
- Added custom input validation to the forex endpoint to reject forex requests if env vars are not set

## Steps to Test

1. Set required env vars for NCFX v3 and not setting the forex env vars
2. Run the adapter
3. Send request for crypto and notice it returns normally
4. Send request for forex and notice it returns immediately with an error

## Quality Assurance
